tests/unit_tests/test_matchmaker_algorithm_team_matchmaker.py

```diff
@@ -0,0 +1,321 @@
+    c1 = CombinedSearch(*[s.pop(), s.pop()])
+    c2 = CombinedSearch(*[s.pop(), s.pop()])
+    s.append(c1)
+    s.append(c2)
+    team_a = CombinedSearch(*[c2, s[0], s[1]])
+    team_b = CombinedSearch(*[c1, s[2], s[3]])
+
+    teams = TeamMatchMaker().make_teams(s, 4)
+
+    assert set(teams[0].players) == set(team_a.players)
+    assert set(teams[1].players) == set(team_b.players)
+    assert abs(teams[0].cumulated_rating - teams[1].cumulated_rating) < 50
+
+
+def test_make_teams_with_negative_rated_players(player_factory):
+    s = make_searches([625, -184, -15, 94, 403, 526, 637, -218], player_factory)
+    c1 = CombinedSearch(*[s.pop(), s.pop()])
+    s.append(c1)
+    team_a = CombinedSearch(*[s[0], s[1], s[3], s[4]])
+    team_b = CombinedSearch(*[s[2], s[5], s[6]])
+
+    teams = TeamMatchMaker().make_teams(s, 4)
+
+    assert set(teams[0].players) == set(team_a.players)
+    assert set(teams[1].players) == set(team_b.players)
+    assert abs(teams[0].cumulated_rating - teams[1].cumulated_rating) < 50
+
+
+def test_make_teams_with_full_sized_search(player_factory):
+    s = make_searches([925, 1084, 1015, 1094, 1403, 1526, 1637, 1718], player_factory)
+    c1 = CombinedSearch(*[s.pop(), s.pop(), s.pop(), s.pop()])
+    s.append(c1)
+    team_a = CombinedSearch(*[c1])
+    team_b = CombinedSearch(*[s[0], s[1], s[2], s[3]])
+
+    teams = TeamMatchMaker().make_teams(s, 4)
+
+    assert set(teams[0].players) == set(team_a.players)
+    assert set(teams[1].players) == set(team_b.players)
+
+
+def test_make_teams_with_almost_full_sized_search(player_factory):
+    s = make_searches([1415, 125, 1294, 584, 1303, 1526, 1237, 1218], player_factory)
+    c1 = CombinedSearch(*[s.pop(), s.pop(), s.pop()])
+    s.append(c1)
+
+    team_a = CombinedSearch(*[c1, s[3]])
+    team_b = CombinedSearch(*[s[0], s[1], s[2], s[4]])
+    teams = TeamMatchMaker().make_teams(s, 4)
+
+    assert set(teams[0].players) == set(team_a.players)
+    assert set(teams[1].players) == set(team_b.players)
+
+
+def test_ignore_impossible_team_splits(player_factory):
+    s = make_searches([1415, 125, 1294, 584, 1303, 1526, 1237, 1218], player_factory)
+    c1 = CombinedSearch(*[s.pop(), s.pop(), s.pop()])
+    c2 = CombinedSearch(*[s.pop(), s.pop(), s.pop()])
+    c3 = CombinedSearch(*[s.pop(), s.pop()])
+    s.append(c1)
+    s.append(c2)
+    s.append(c3)
+
+    matches, unmatched = TeamMatchMaker().find(s, 4)
+
+    assert len(matches) == 0
+    assert set(unmatched) == set(s)
+
+
+@given(team_a=st_searches(4), team_b=st_searches(4))
+def test_game_quality(team_a, team_b):
+    game = TeamMatchMaker().assign_game_quality((team_a, team_b))
+
+    assert 0.0 <= game.quality <= 1.0
+
+
+@given(player=st_players())
+def test_maximum_game_quality_for_even_teams(player):
+    search = Search([player])
+    team_a = CombinedSearch(*[search, search, search, search])
+    team_b = CombinedSearch(*[search, search, search, search])
```

Comment:
In python you can multiply lists by integers:
```suggestion
    team_a = CombinedSearch(*[search] * 4)
    team_b = CombinedSearch(*[search] * 4)
```

Reply:
Wait, wat? Python never seizes to amaze...